popout.js

```diff
@@ -1,241 +1,355 @@
+            }, this.TIMEOUT_INTERVAL);
+            return;
+        }
+
+        if (this.poppedOut.has(app.appId)) {
+            this.log("Already popped out");
+            this.poppedOut.get(app.appId).focus();
+            return;
+        }
+
+
+        if (app && app.options && app.options.classes.includes("dialog")) {
+            this.log("is dialog");
+            if (app.actor && app.actor.apps) {
+                const keys = Object.keys(app.actor.apps);
+                this.log("has an apps keys", keys);
+                if (keys.length == 1) {
+                    const parent = app.actor.apps[keys[0]]
+                    if (this.poppedOut.has(parent.appId)) {
+                        this.log("Intercepting dialog of popped out window.");
+                        this.moveDialog(app, parent);
+                        return;
+                    }
+                }
+            }
+        }
+
+        const domID = `popout_${this.ID}_${app.appId}`;
+        if (!document.getElementById(domID)) { // Don't create a second link on re-renders;
+            const link = $(`<a id="${domID}"><i class="fas fa-external-link-alt"></i>Popout</a>`)
+            link.on("click", () => this.onPopoutClicked(app));
+            const title = app.element.find(".window-title").after(link);
+            this.log("Attached", app);
+        }
+    }
+
+    moveDialog(app, parent) {
+        const parentWindow = this.poppedOut.get(parent.appId);
+        const dialogNode = app.element[0];
+
+        // Hide element
+        const setDisplay = dialogNode.style.display;
+        dialogNode.style.display = "None";
+
+        const newHeader = parentWindow.document.createElement("header")
+        newHeader.setAttribute("class", "window-header flexrow");
+        const headerElements = dialogNode.children[0].children;
+        for (const element of Array.from(headerElements)) {
+            newHeader.appendChild(parentWindow.document.adoptNode(element));
+        }
+
+        dialogNode.children[0].remove()
+
+        const node = parentWindow.document.adoptNode(dialogNode);
+        node.style.top = "50%";
+        node.style.left = "50%";
+        node.style.transform = "translate(-50%, -50%)"
+
+        node.insertBefore(newHeader, node.children[0]);
+
+        parent.element[0].parentNode.insertBefore(node, parent.element[0].nextSibling);
+        node.style.display = setDisplay;
+    }
+
+
+    onPopoutClicked(app) {
+        // Check if popout in Electron window
+        if (navigator.userAgent.toLowerCase().indexOf(" electron/") !== -1) {
+            ui.notifications.warn("Popout! cannot work within the standalone FVTT Application. Please open your game from a regular browser.");
+            return;
+        }
+
+        // -------------------- Obtain application --------------------
+
+        // Store original position for later use.
+        const appPosition = { ...app.position };

```

Comment:
I was using the spread operator because I am used to it, I have changed it to duplicate method to stick with the foundry js style. (Also the spread operator is only a shallow copy).

Reply:
Ah yeah, shallow vs deep copy, you're right! In the case of `position`, it wouldn't matter, but it's good to know/keep in mind for the future. Thanks!